internal/abci/abci.go

```diff
@@ -820,13 +848,17 @@ func (a *AbciApp) FinalizeBlock(ctx context.Context, req *abciTypes.RequestFinal
 		}
 		if err = a.p2p.RemovePeer(ctx, addr); err != nil {
 			if !errors.Is(err, cometbft.ErrPeerNotWhitelisted) {
-				return nil, fmt.Errorf("failed to remove expired validator %s from peer list: %w", addr, err)
+				a.log.Warn("failed to remove expired validator from peer list", zap.String("address", addr), zap.Error(err))
 			}
 		}
 	}
 
 	// wait for all logs to be received
-	wg.Wait()
+	select {
+	case <-doneLogs:
+	case <-ctx.Done():
```

Comment:
Well crud, this isn't working as I expected.  If I remove the `case <-doneLogs`, and shutdown PostgreSQL, it does stop everything, but comet doesn't cancel the context it passed:

```
2024-09-24T17:52:39.35-05:00	info	kwild.server	stop listeners
2024-09-24T17:52:39.35-05:00	info	kwild.server	stop comet server
2024-09-24T17:52:39.35-05:00	info	kwild.admin-jsonrpc-server	JSON-RPC server shutting down...
2024-09-24T17:52:39.35-05:00	info	kwild.listener-manager	ListenerManager stopped.
2024-09-24T17:52:39.35-05:00	info	kwild.user-jsonrpc-server	JSON-RPC server shutting down...
2024-09-24T17:52:39.35-05:00	info	kwild.cometbft	service stop	{"msg": "Stopping Node service", "impl": "Node"}
2024-09-24T17:52:39.35-05:00	info	kwild.user-jsonrpc-server	JSON-RPC listener done for [::]:8484
2024-09-24T17:52:39.35-05:00	info	kwild.cometbft	Stopping Node
2024-09-24T17:52:39.35-05:00	info	kwild.user-jsonrpc-server	JSON-RPC server shutdown complete
2024-09-24T17:52:39.35-05:00	info	kwild.admin-jsonrpc-server	JSON-RPC listener done for /tmp/kwild.socket
2024-09-24T17:52:39.35-05:00	info	kwild.admin-jsonrpc-server	JSON-RPC server shutdown complete
2024-09-24T17:52:39.35-05:00	info	kwild.cometbft	service stop	{"module": "events", "msg": "Stopping EventBus service", "impl": "EventBus"}
2024-09-24T17:52:39.35-05:00	info	kwild.cometbft	service stop	{"module": "events", "module": "pubsub", "msg": "Stopping PubSub service", "impl": "PubSub"}
2024-09-24T17:52:39.35-05:00	info	kwild.cometbft	service stop	{"module": "txindex", "msg": "Stopping IndexerService service", "impl": "IndexerService"}
2024-09-24T17:52:39.35-05:00	info	kwild.cometbft	service stop	{"module": "p2p", "msg": "Stopping P2P Switch service", "impl": "P2P Switch"}
2024-09-24T17:52:39.35-05:00	info	kwild.cometbft	service stop	{"module": "pex", "msg": "Stopping PEX service", "impl": "PEX"}
2024-09-24T17:52:39.35-05:00	info	kwild.cometbft	service stop	{"module": "p2p", "book": "/home/jon/.kwild-junk3/abci/addrbook.json", "msg": "Stopping AddrBook service", "impl": "AddrBook"}
2024-09-24T17:52:39.35-05:00	info	kwild.cometbft	service stop	{"module": "mempool", "msg": "Stopping Mempool service", "impl": "Mempool"}
2024-09-24T17:52:39.35-05:00	info	kwild.cometbft	Saving AddrBook to file	{"module": "p2p", "book": "/home/jon/.kwild-junk3/abci/addrbook.json", "size": 0}
2024-09-24T17:52:39.35-05:00	info	kwild.cometbft	service stop	{"module": "blocksync", "msg": "Stopping Reactor service", "impl": "Reactor"}
2024-09-24T17:52:39.35-05:00	info	kwild.cometbft	service stop	{"module": "consensus", "msg": "Stopping Consensus service", "impl": "ConsensusReactor"}
2024-09-24T17:52:39.35-05:00	info	kwild.cometbft	service stop	{"module": "consensus", "msg": "Stopping State service", "impl": "ConsensusState"}
2024-09-24T17:52:39.35-05:00	info	kwild.cometbft	service stop	{"module": "consensus", "msg": "Stopping TimeoutTicker service", "impl": "TimeoutTicker"}
```

Rethinking this...

Reply:
OKay, I am gonna remove these changes from this PR.